src/main/java/fi/dy/masa/tweakeroo/util/InventoryUtils.java

```diff
@@ -325,24 +333,26 @@ public static void trySwitchToEffectiveTool(BlockPos pos)
 
         if (player != null && mc.world != null)
         {
-            BlockState state = mc.world.getBlockState(pos);
-            ScreenHandler container = player.playerScreenHandler;
-            ItemPickerTest test;
+            if (!TOOL_SWITCH_IGNORED_SLOTS.contains( player.getInventory().selectedSlot)){
```

Comment:
I would put this check in the same if-block above, as the last condition (especially to avoid an unnecessary indentation change of the entire block).
Also follow my "code style" here and use
```java
if (player != null && mc.world != null &&
    TOOL_SWITCH_IGNORED_SLOTS.contains(player.getInventory().selectedSlot == false)
```